ts-defs/index.d.ts

```diff
@@ -1543,6 +1543,194 @@ interface Assertion {
+    /**
+     * Creates the test runner that is used to configure and launch test tasks.
+     */
+    createRunner(): Runner;
+
+    /**
+     * Creates a remote browser connection.
+     */
+    createBrowserConnection(): Promise<BrowserConnection>;
+
+    /**
+     * Stops the TestCafe server. Forcibly closes all connections and pending test runs immediately.
+     */
+    close(): void;
+}
+
+interface Runner {
+    /**
+     * Configures the test runner to run tests from the specified files.
+     *
+     * @param source - The relative or absolute path to a test fixture file, or several such paths. You can use glob patterns to include (or exclude) multiple files.
+     */
+    src(source: string | string[]): this;
+
+    /**
+     * Allows you to select which tests should be run.
+     *
+     * @param callback - The callback that determines if a particular test should be run.
+     * @param callback `testName` - The name of the test.
+     * @param callback `fixtureName` - The name of the test fixture.
+     * @param callback `fixturePath` - The path to the test fixture file.
+     * @param callback `testMeta` - The test metadata.
+     * @param callback `fixtureMeta` - The fixture metadata.
+     */
+    filter(
+        callback: (
+        testName: string,
+        fixtureName: string,
+        fixturePath: string,
+        testMeta: Record<string, string>,
+        fixtureMeta: Record<string, string>
+        ) => boolean
+    ): this;
+
+    /**
+     * Configures the test runner to run tests in the specified browsers.
+     *
+     * @param browser - A different browser alias for each browser type.
+     */
+    browsers(browser: string | string[]): this;
+
+    /**
+     * Configures the test runner to run tests in the specified browsers.
+     *
+     * @param browser - The path to the browser's executable (path) and command line parameters (cmd).
+     */
+    browsers(browser: { path: string; cmd?: string }): this;
+
+    /**
+     * Configures the test runner to run tests in the specified browsers.
+     *
+     * @param browser - The remote browser connection.
+     */
+    browsers(browser: BrowserConnection): this;
+
+    /**
+     * Enables TestCafe to take screenshots of the tested webpages.
+     *
+     * @param path - The base path where the screenshots are saved. Note that to construct a complete path to these screenshots, TestCafe uses default path patterns.
+     * @param takeOnFails - Specifies if screenshots should be taken automatically when a test fails.
+     * @param pathPattern - The pattern to compose screenshot files' relative path and name.
+     */
+    screenshots(path: string, takeOnFails?: boolean, pathPattern?: string): this;
+
+    /**
+     * Configures TestCafe's reporting feature.
+     *
+     * @param name - The name of the reporter to use.
+     * @param outStream - The stream to which the report is written.
+     */
+    reporter(name: string, outStream?: NodeJS.WritableStream): this;
```

Comment:
I've missed that we have recently added a new overload of the `reporter` method'. Summing up with the old `reporter` method:
```
reporter(name: string, file?: NodeJS.WritableStream): this;
reporter([reporter: string | { name: string, file?: string | NodeJS.WritableStream }]): this;
```

Reply:
Added overloads and tests, have a look at it, not sure what should be tested here tho